can/io/asc.py

```diff
@@ -1,7 +1,49 @@
 from can.listener import Listener
+from can.message import Message
 
 from datetime import datetime
 import time
+CAN_MSG_EXT = 0x80000000
+
+
+class ASCReader(object):
+    """
+    Iterator of CAN messages from a ASC Logging File.
+
+    Only CAN messages are supported (no Errorframes).
+    """
+
+    def __init__(self, filename):
+        self.fp = open(filename, "r")
+
+    def __iter__(self):
+        for line in self.fp:
+            temp = line.strip()
+            if len(temp) > 0:
+                if temp[0].isdigit():
+                    lineArray = temp.split()
+                    if lineArray[1].isdigit() and lineArray[2] != "Statistic:":
+                        time = float(lineArray[0])
+                        channel = lineArray[1]
+                        can_id = int(lineArray[2],16)
```

Comment:
Extended frames have an x suffixed after the CAN ID (see [here](http://zone.ni.com/reference/en-XX/help/370859J-01/dlgcanconverter/dlgcanconverter/canconverter_ascii_logfiles/)), so this will probably fail.

Have a look at listener_test.py and BLFTest and make a test that writes an ASC log file and then reads it back again.

Reply:
Thanks for the feedback.
I'll add extended support for IDs and add some documentation and tests.